task/transcode.go

```diff
@@ -116,6 +117,22 @@ func fileWriter(size int64) WriteSeekCloser {
 	}
 }
 
+type SegmentSizeAccumulator struct {
```

Comment:
nit: So... Nothing in this interface assumes or required that the thing being accumulated is the segment size. It can be used to accumulate any other value, like for example the read bytes in the case of the read counter.

I believe this be just called Accumulator. It can also be used from the ReadCounter above